crates/dojo-core/src/world.cairo

```diff
@@ -564,6 +565,32 @@ mod world {
         }
     }
 
+    #[abi(embed_v0)]
+    impl UpgradeableState of IUpgradeableState<ContractState> {
+        fn upgrade_state(ref self: ContractState, new_state: Span<felt252>, program_output: Span<felt252>) {
```

Comment:
```suggestion
        struct NewStatePair {
            storage_key: felt252,
            value: felt252,
        }
        
        fn upgrade_state(ref self: ContractState, new_state: Span<NewStatePair>, program_output: ProgramOutput) {
```
We should use Cairo `serde` as much as we can that now we have it out of the box! It may also make the code easier to read and manageable while it's growing.